pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,9 +70,117 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
+		Parameters: map[string]string{
+			"objectStoreName":      fmt.Sprintf("%s-object-store", initData.Namespace),
+			"objectStoreNamespace": initData.Namespace,
+			"region":               "us-east-1", // TODO: how to get the current region
+		},
+	}
+	return retSC
+}
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResources
+	err = json.Unmarshal(found.Data[externalClusterSecretDataKeyName], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace given for ConfigMap?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			cm.Data = d.Data
+			found := &corev1.ConfigMap{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph ConfigMap:	" + cm.Name)
+					err = r.client.Create(context.TODO(), cm)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "Secret":
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace for Secret?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			for k, v := range d.Data {
+				s.Data[k] = []byte(v)
+			}
+			found := &corev1.Secret{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph Secret: 	" + s.Name)
+					err = r.client.Create(context.TODO(), s)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "StorageClass":
+			scs, err := r.newStorageClasses(instance)
+			if err != nil {
+				return err
+			}
+			scCephFs := scs[0]
+			scCephFs.Parameters["fsName"] = d.FilesystemName
+			scRBD := scs[1]
+			scRBD.Parameters["pool"] = d.PoolName
```

Comment:
This is part of the Provisioner Secret not the StorageClass. See the JSON blob:

```
{
    "kind": "Secret",
    "data": {
      "userKey": "AQBU66peVcBCKxAAoPQ4tKlJ9gxvrH3MRnDRHQ==",
      "userID": "csi-rbd-provisioner"
    },
    "name": "rook-csi-rbd-provisioner",
    "pool-name": "rbd-pool"
}
```

Reply:
I didn't quite understand what you meant by `Provisioner Secret` here?. I may be wrong here but, how are we going to differentiate if it's "Secret" or a "Provisioner Secret"?.

Reply:
We just moved these into its own StorageClass section, let me comment your code again

Reply:
Now, this is d.Data["pool-name"]